### PR TITLE
feat: pagination could change its size by the width of the window

### PR DESCRIPTION
**Pull request by wzhudev**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#21885

### 💡 Background and solution

如果用户没有指定 pagination 的 size，且传递了 autoResize 时，当屏幕缩小到 xs，会自动呈现为 small 样式。
为了让用户能够在小屏幕上使用默认样式，size 参数新增 `default` 类型。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Pagination 新增 autoResize 属性 |
| 🇨🇳 Chinese | Pagination supports autoResize prop |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed
- [x] Test is provided

-----
[View rendered components/pagination/index.en-US.md](https://github.com/wendellhu95/ant-design/blob/pagination-auto-small/components/pagination/index.en-US.md)
[View rendered components/pagination/index.zh-CN.md](https://github.com/wendellhu95/ant-design/blob/pagination-auto-small/components/pagination/index.zh-CN.md)

### CODE REVIEW

**Comment by afc163** [comment]
别自动，来一个开关属性，默认关闭。

File: components/pagination/Pagination.tsx [hunk 1]
```diff
@@ -26,7 +27,8 @@ export interface PaginationProps {
   onShowSizeChange?: (current: number, size: number) => void;
   showQuickJumper?: boolean | { goButton?: React.ReactNode };
   showTotal?: (total: number, range: [number, number]) => React.ReactNode;
-  size?: string;
+  size?: 'default' | 'small';
+  autoResize?: boolean;
```
Comment (afc163) [inline components/pagination/Pagination.tsx]: responsive

File: components/pagination/Pagination.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ export default class Pagination extends React.Component<PaginationProps, {}> {
   componentDidMount() {
     this.token = ResponsiveObserve.subscribe(screens => {
       const { xs } = screens;
-      const { size, autoResize } = this.props;
+      const { size, responsive: autoResize } = this.props;
```
Comment (afc163) [inline components/pagination/Pagination.tsx]:
```suggestion
      const { size, responsive } = this.props;
```

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
ci broken

**Comment by wendellhu95** [comment]
Fixed.

**Pull request merged**